identity-account/src/tests/account.rs

```diff
@@ -322,3 +335,172 @@ async fn test_account_publish_options_force_integration() -> Result<()> {
+      assert!(account.fetch_state().await.is_err());
+
+      // Case 1: Tangle and account are synched
+      account.publish().await.unwrap();
+      let old_state: IdentityState = account.state().clone();
+      let old_chain_state: ChainState = account.chain_state().clone();
+      account.fetch_state().await.unwrap();
+      assert_eq!(old_state.document(), account.state().document());
+      assert_eq!(&old_chain_state, account.chain_state());
+
+      // Case 2: Local state is ahead of the tangle
+      account
+        .update_identity()
+        .create_service()
+        .fragment("my-other-service")
+        .type_("LinkedDomains")
+        .endpoint(Url::parse("https://example.org").unwrap())
+        .apply()
+        .await?;
+      let old_state: IdentityState = account.state().clone();
+      let old_chain_state: ChainState = account.chain_state().clone();
+      account.fetch_state().await.unwrap();
+      assert_eq!(old_state.document(), account.state().document());
+      assert_eq!(&old_chain_state, account.chain_state());
+      Ok(())
+    })
+  })
+  .await?;
+  Ok(())
+}
+
+#[tokio::test]
+async fn test_account_sync_integration_msg_update() -> Result<()> {
+  network_resilient_test(2, |n| {
+    Box::pin(async move {
+      let network = if n % 2 == 0 { Network::Devnet } else { Network::Mainnet };
+      let config = AccountConfig::default().autopublish(false);
+      let account_config = AccountSetup::new_with_options(
+        Arc::new(MemStore::new()),
+        Some(config),
+        Some(Arc::new(ClientMap::from_network(network.clone()).await.unwrap())),
+      );
+      let mut account =
+        Account::create_identity(account_config, IdentitySetup::new().network(network.name()).unwrap()).await?;
+      account.publish().await.unwrap();
+
+      let client: Client = Client::from_network(network).await.unwrap();
+      let mut new_doc: IotaDocument = account.document().clone();
+      new_doc.properties_mut().insert("foo".into(), 123.into());
+      new_doc.properties_mut().insert("bar".into(), 456.into());
+      new_doc.metadata.previous_message_id = *account.chain_state().last_integration_message_id();
+      new_doc.metadata.updated = Timestamp::now_utc();
+      account
+        .sign(IotaDocument::DEFAULT_METHOD_FRAGMENT, &mut new_doc)
+        .await
+        .unwrap();
+      client.publish_document(&new_doc).await.unwrap();
+      let chain: DocumentChain = client.read_document_chain(account.did()).await.unwrap();
+
+      account.fetch_state().await.unwrap();
+      assert!(account.state().document().properties().contains_key("foo"));
+      assert!(account.state().document().properties().contains_key("bar"));
+      assert_eq!(
+        account.chain_state().last_integration_message_id(),
+        chain.integration_message_id()
+      );
+      assert_eq!(account.chain_state().last_diff_message_id(), chain.diff_message_id());
+      Ok(())
```

Comment:
```suggestion
      assert_eq!(account.chain_state().last_diff_message_id(), chain.diff_message_id());
      
      // Ensure state was written into storage.
      let storage_state: IdentityState = account.load_state().await?;
      assert_eq!(&storage_state, account.state());
      
      Ok(())
```

Test that fetched state was written into storage. Maybe also set `AutoSave::Every` explicitly in the `AccountConfig` (line 394) to make it explicit that this test requires that setting.